packages/schematics/angular/migrations/update-8/update-lazy-module-paths.ts

```diff
@@ -5,26 +5,76 @@
  * Use of this source code is governed by an MIT-style license that can be
  * found in the LICENSE file at https://angular.io/license
  */
-import {
-  Rule,
-  SchematicContext,
-  Tree,
-} from '@angular-devkit/schematics';
-import {
-  TslintFixTask,
-} from '@angular-devkit/schematics/tasks';
-import * as path from 'path';
-
-export const updateLazyModulePaths = (): Rule => {
-  return (_: Tree, context: SchematicContext) => {
-    context.addTask(new TslintFixTask({
-      rulesDirectory: path.join(__dirname, 'rules'),
-      rules: {
-        'no-lazy-module-paths': [true],
-      },
-    }, {
-      includes: '**/*.ts',
-      silent: false,
-    }));
+import { DirEntry, Rule, UpdateRecorder } from '@angular-devkit/schematics';
+import * as ts from '../../third_party/github.com/Microsoft/TypeScript/lib/typescript';
+
+function* visit(directory: DirEntry): IterableIterator<ts.SourceFile> {
+  for (const path of directory.subfiles) {
+    if (path.endsWith('.ts') && !path.endsWith('.d.ts')) {
+      const entry = directory.file(path);
+      if (entry) {
+        const content = entry.content;
+        if (content.includes('loadChildren')) {
+          const source = ts.createSourceFile(
+            entry.path,
+            content.toString(),
+            ts.ScriptTarget.Latest,
+            true,
+          );
+
+          yield source;
+        }
+      }
+    }
+  }
+
+  for (const path of directory.subdirs) {
+    if (path === 'node_modules') {
+      continue;
+    }
+
+    yield* visit(directory.dir(path));
+  }
+}
+
+export function updateLazyModulePaths(): Rule {
+  return tree => {
+    for (const sourceFile of visit(tree.root)) {
+      let recorder: UpdateRecorder | undefined;
+
+      ts.forEachChild(sourceFile, function analyze(node) {
+        if (ts.isPropertyAssignment(node) &&
+            (ts.isIdentifier(node.name) || ts.isStringLiteral(node.name)) &&
+            node.name.text === 'loadChildren' &&
+            ts.isStringLiteral(node.initializer)) {
+          const valueNode = node.initializer;
+          const parts = valueNode.text.split('#');
+          const path = parts[0];
+          const moduleName = parts[1] || 'default';
+
+          let fix = `() => import('${path}').then(m => m.${moduleName})`;
+
+          // Try to fix indentation in replacement:
+          const { character } = sourceFile.getLineAndCharacterOfPosition(node.getStart());
+          fix = fix.replace(/\n/g, `\n${' '.repeat(character)}`);
```

Comment:
`fix` has no `\n`, I don't think this'll ever replace anything.

Reply:
Removed.